src/pages/gifts/claim/[guid].tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from 'react'
+import {GetServerSideProps} from 'next'
+import {getTokenFromCookieHeaders} from '../../../utils/auth'
+import {getGift} from '../../../lib/gifts'
+import Link from 'next/link'
+import axios from 'utils/configured-axios'
+import {useRouter} from 'next/router'
+import {useViewer} from '../../../context/viewer-context'
+import {PrimaryButton} from '../../../components/buttons'
+
+// tachyons to tailwind converter https://gist.github.com/knoopx/240cbe3b16a7a2f9e4101238533f87c8
+
+export const getServerSideProps: GetServerSideProps = async function ({
+  req,
+  query,
+}) {
+  const {eggheadToken} = getTokenFromCookieHeaders(req.headers.cookie as string)
+  const {guid} = query
+
+  const gift = await getGift(guid as string, eggheadToken)
+
+  if (!eggheadToken) {
+    return {
+      props: {
+        error: 'You must be logged in to claim a membership token',
+      },
+    }
+  }
+
+  if (gift.claimed) {
+    return {
+      props: {
+        error: 'This membership token has already been claimed',
+      },
+    }
+  }
+
+  return {
+    props: {gift},
+  }
+}
+
+type Gift = {claim_url: string; claimed: boolean; duration_months: number}
+type GiftClaimProps = {error?: string; gift?: Gift}
+
+const GiftClaim: React.FC<GiftClaimProps> = ({error, gift}) => {
+  const router = useRouter()
+  const {viewer, refreshUser} = useViewer()
+
+  const handleClaimGift = async () => {
+    if (!gift) {
+      return
+    }
+    const {claim_url} = gift
+
+    await axios.post(claim_url)
+
+    refreshUser()
+
+    router.push('/')
```

Comment:
the viewer/user doesn't get refreshed here and there is no messaging that it succeeded